### DISCUSSION TITLE
FulfillmentInbound and CurrenyCode Error

### DISCUSSION THREAD

**Discussion by IncredibleX** [abuzuhri/Amazon-SP-API-CSharp#146] [Q&A]
Hi,
while i try to create an inboundshipmentplan
i got the following message:
Error converting value "EUR" to type 'FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.CurrencyCode'. Path 'payload.InboundShipmentPlans[0].EstimatedBoxContentsFee.FeePerUnit.CurrencyCode', line 51, position 67.

Any idea?
Thanx,
Tobias

**Comment by abuzuhri** [comment]
Hi 

Can you share the code you try to use ?

**Reply by IncredibleX** [reply]
`public void CreateShipmentPlan(string Sendershortname)
            {
                FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.CreateInboundShipmentPlanRequest oCreateInboundShipmentPlanRequest = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.CreateInboundShipmentPlanRequest();

                oCreateInboundShipmentPlanRequest.ShipToCountryCode = "DE";
                oCreateInboundShipmentPlanRequest.LabelPrepPreference = FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.LabelPrepPreference.SELLERLABEL;


                eXODALib.AddressLib.AddressOrganisation oAdrOrga = new AddressLib.AddressOrganisation();
                oAdrOrga.Initialize(exSettings);
                eXODALib.AddressLib.Address oAdr = new AddressLib.Address();
                oAdr = oAdrOrga.Open(Sendershortname);
                oCreateInboundShipmentPlanRequest.ShipFromAddress = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.Address();
                oCreateInboundShipmentPlanRequest.ShipFromAddress.AddressLine1 = oAdr.Name2;
                oCreateInboundShipmentPlanRequest.ShipFromAddress.AddressLine2 = oAdr.Street;
                oCreateInboundShipmentPlanRequest.ShipFromAddress.City = oAdr.City;
                oCreateInboundShipmentPlanRequest.ShipFromAddress.CountryCode = "DE";
                oCreateInboundShipmentPlanRequest.ShipFromAddress.PostalCode = oAdr.Postcode;
                oCreateInboundShipmentPlanRequest.ShipFromAddress.Name = oAdr.Name1;

                

                oCreateInboundShipmentPlanRequest.InboundShipmentPlanRequestItems = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.InboundShipmentPlanRequestItemList();
                FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.InboundShipmentPlanRequestItem oInboundShipmentPlanRequestItem = new FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.InboundShipmentPlanRequestItem();
                oInboundShipmentPlanRequestItem.SellerSKU = "16118";
                oInboundShipmentPlanRequestItem.ASIN = "B08BXH6234";
                oInboundShipmentPlanRequestItem.Quantity = 1;
                oInboundShipmentPlanRequestItem.Condition = FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.Condition.NewItem;

                oCreateInboundShipmentPlanRequest.InboundShipmentPlanRequestItems.Add(oInboundShipmentPlanRequestItem);
                FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.CreateInboundShipmentPlanResult oResult = amazonConnection.FulFillmentInbound.CreateInboundShipmentPlan(oCreateInboundShipmentPlanRequest);

            }`

**Reply by IncredibleX** [reply]
Hi @abuzuhri ,
my solution is to change the current culture in my Applicition:
Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US"); 

This works...

**Comment by IncredibleX** [comment]
Hmm...i thought this was the solution - i also added following enum code EURO in your CurrencyCodes.cs:

Can you add this in your Library?

public enum CurrencyCode
    {

        /// <summary>
        /// Enum USD for value: USD
        /// </summary>
        [EnumMember(Value = "USD")]
        USD = 1,

        /// <summary>
        /// Enum GBP for value: GBP
        /// </summary>
        [EnumMember(Value = "GBP")]
        GBP = 2,
        /// <summary>
        /// Enum EUR for value: EUR
        /// </summary>
        [EnumMember(Value = "EUR")]
        EUR = 3
    }

**Reply by abuzuhri** [reply]
Ok I will do it today

**Reply by abuzuhri** [reply]
Whe

> Hmm...i thought this was the solution - i also added following enum code EURO in your CurrencyCodes.cs:
> 
> Can you add this in your Library?
> 
> public enum CurrencyCode {
> 
> ```
>     /// <summary>
>     /// Enum USD for value: USD
>     /// </summary>
>     [EnumMember(Value = "USD")]
>     USD = 1,
> 
>     /// <summary>
>     /// Enum GBP for value: GBP
>     /// </summary>
>     [EnumMember(Value = "GBP")]
>     GBP = 2,
>     /// <summary>
>     /// Enum EUR for value: EUR
>     /// </summary>
>     [EnumMember(Value = "EUR")]
>     EUR = 3
> }
> ```

Where you use currency in your code i cant see the currency

**Reply by abuzuhri** [reply]
> Hmm...i thought this was the solution - i also added following enum code EURO in your CurrencyCodes.cs:
> 
> Can you add this in your Library?
> 
> public enum CurrencyCode {
> 
> ```
>     /// <summary>
>     /// Enum USD for value: USD
>     /// </summary>
>     [EnumMember(Value = "USD")]
>     USD = 1,
> 
>     /// <summary>
>     /// Enum GBP for value: GBP
>     /// </summary>
>     [EnumMember(Value = "GBP")]
>     GBP = 2,
>     /// <summary>
>     /// Enum EUR for value: EUR
>     /// </summary>
>     [EnumMember(Value = "EUR")]
>     EUR = 3
> }
> ```

https://github.com/amzn/selling-partner-api-docs/blob/76230f1ce525150622460af572c3a1c9d94b0b9d/references/fulfillment-inbound-api/fulfillmentInboundV0.md#currencycode

**Comment by IncredibleX** [comment]
Thanx!

Tobias Stephan
eCommerce & Softwaredevelopment
+49 171 3833568
***@***.***
Shops:
www.leckerstecker.de
www.herr-der-dinge.de

Softwareblog:
www.exoda.de
Kookamp 40, 46354 Südlohn
Apps:
Leckerstecker Kabelshop
3D Print for Dremel 3D45 3D40
Listory Lister Shopping App
MCR-Tools

...noch mehr

Am 23. März 2022, 19:36 +0100 schrieb Tareq Abuzuhri ***@***.***>:

**Comment by IncredibleX** [comment]
Hi,

you are right - i don’t use the currency, but anyway i got the Errormessage:
Error converting value "EUR" to type 'FikaAmazonAPI.AmazonSpApiSDK.Models.FulfillmentInbound.CurrencyCode'. Path

So i added quick an dirty the enum value EUR to your file - after that it worked!

Tobias Stephan
eCommerce & Softwaredevelopment
+49 171 3833568
***@***.***
Shops:
www.leckerstecker.de
www.herr-der-dinge.de

Softwareblog:
www.exoda.de
Kookamp 40, 46354 Südlohn
Apps:
Leckerstecker Kabelshop
3D Print for Dremel 3D45 3D40
Listory Lister Shopping App
MCR-Tools

...noch mehr

Am 24. März 2022, 15:30 +0100 schrieb Tareq Abuzuhri ***@***.***>:
